pkg/flow/flow_http.go

```diff
@@ -43,6 +44,35 @@ func (f *Flow) ConfigHandler() http.HandlerFunc {
 	}
 }
 
+// ComponentHandler returns an http.HandlerFunc which will delegate all requests to
+// a component named by the first path segment
+func (f *Flow) ComponentHandler() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		parts := strings.Split(r.URL.Path, "/")
+		id := parts[1]
+		newPath := "/" + strings.Join(parts[2:], "/")
```

Comment:
I find this a little hard to follow and reason about. Can we take more advantage of mux routing featuers here to make it easier to understand? Something like using `/component/{componentID}` in the route name and potentially a middleware handler to trim the prefix?

Reply:
yeah, I agree its ugly. I'll hit their docs and see the right way to do it. Trickiest part is grabbing the entire trimmed subpath and passing it along as the new root.